scripts/tests_analyzer/compare_coderabbit_decisions.py

```
@@ -0,0 +1,689 @@
+        results.append(result)
+
+    # Generate output
+    if args.output == "json":
+        output = json.dumps([r.to_dict() for r in results], indent=2)
+    else:
+        output = generate_markdown_report(results=results, repo=args.repo, detailed=args.detailed)
+
+    # Write output
+    if args.output_file:
+        args.output_file.write_text(data=output)
+        logger.info(f"Report written to {args.output_file}")
```

Ruff (0.14.10)

Logging statement uses f-string

(G004)